src/Microsoft.IdentityModel.Tokens/TokenValidationParameters.cs

```diff
@@ -96,6 +96,12 @@ namespace Microsoft.IdentityModel.Tokens
     public delegate SecurityToken SignatureValidator(string token, TokenValidationParameters validationParameters);
 
     /// <summary>
+    /// Definition for TokenReader.
+    /// </summary>
+    /// <param name="token">A securityToken with a signature.</param>
+    public delegate SecurityToken TokenReader(string token);
```

Comment:
Even though it is not needed, we always pass TokenValidationParameters to facilitate extensibility.

Reply:
Got it!